sql-plugin-api/src/main/scala/com/nvidia/spark/rapids/RapidsProxy.scala

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+/**
+ * Trait to replace usage of deprecated scala.Proxy. If the purpose is to delay
+ * initialization of the underlying implementation, self is overriden as a
+ * lazy val
+ */
+trait RapidsProxy extends Any {
+  def self: Any
+}
```

Comment:
I'm with @tgravescs in that I don't see the point of this class.  It doesn't seem help with method delegation in any meaningful way.  We still have to explicitly declare the underlying instance and explicitly declare each method that will be delegated to that instance.

Reply:
scala.Proxy was deprecated in 2.13 because it creates more problems than it solves, and thus I'm wondering why we would want it when Scala has decided it's not worth it.  Personally I don't see the value of even tagging proxies, as there's other cases where we sometimes delegate, sometimes don't -- are those Proxy or not?  I'm guessing not.

Even with the trivial version that is here, which does basically nothing, it encourages people to define `self` as a `def` when it almost never should be a `def` in practice.  If we want a trait just for tagging, fine, not sure I care enough either way.  However I would just have it be a trait with no implementation, as currently I think the required `self` definition causes more problems than it solves.